client/layout/guided-tours/tours/marketing-connections-tour/index.js

```diff
@@ -10,6 +10,16 @@ import meta from './meta';
 import { ButtonRow, makeTour, Quit, Link, Step, Tour } from 'layout/guided-tours/config-elements';
 import { localizeUrl } from 'lib/i18n-utils';
 
+const CONNECT_BUTTON_SELECTOR = '.sharing-service.not-connected .button.is-compact';
+
+// Wait until the desired DOM element appears. Check every 125ms.
+// This function is a Redux action creator, hence the two arrows.
+const waitForConnectButton = () => async () => {
+	while ( ! document.querySelector( CONNECT_BUTTON_SELECTOR ) ) {
+		await new Promise( ( resolve ) => setTimeout( resolve, 125 ) );
+	}
+};
+
```

Comment:
This seems like a hacky solution, but it is [a pattern used elsewhere](https://github.com/Automattic/wp-calypso/blob/master/client/layout/guided-tours/tours/jetpack-plugin-updates-tour/index.tsx#L27) so perhaps it's ok for now here too?